cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java

```diff
@@ -1843,6 +1847,7 @@ private void dohandle(Callback[] callbacks) throws IOException, UnsupportedCallb
                     } else {
                         final String defaultName = ncb.getDefaultName();
                         if (defaultName != null) {
+                            this.username = defaultName;
```

Comment:
I am not 100% sure on this yet.  this.userName was intended to hold the username passed in from the command line.

If I am reading this correctly the default name could come from the authentication config and then get set here, the problem here is the CLI could then connect to a different server so resolve a different authentication configuration but the username has now been set based on the first match.